### PR TITLE
Add descriptions to api/extensions/v1beta1/types.go

### PR DESCRIPTION
**Pull request by wangyysde**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Ref #99675

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by wangyysde** [comment]
@ncdc

**Comment by wangyysde** [comment]
@mwielgus

File: staging/src/k8s.io/api/extensions/v1beta1/types.go [hunk 1]
```diff
@@ -729,6 +732,13 @@ type IngressRuleValue struct {
 	// 2. Consider adding fields for ingress-type specific global options
 	// usable by a loadbalancer, like http keep-alive.
 
+	// HTTP is a pointer pointing to a list of http selectors.
+	// The list is a collection of paths that map requests to backends.
+	// A path is matched against the path of an incoming request. Currently it can
+	// contain characters disallowed from the conventional "path" part of a URL
+	// as defined by RFC 3986. Paths must begin with a '/'.
+	// A backend defines the referenced service endpoint to which the traffic
+	// will be forwarded to.
```
Comment (liggitt) [inline staging/src/k8s.io/api/extensions/v1beta1/types.go:741]: cc @robscott for review

File: staging/src/k8s.io/api/extensions/v1beta1/types.go [hunk 2]
```diff
@@ -729,6 +732,13 @@ type IngressRuleValue struct {
 	// 2. Consider adding fields for ingress-type specific global options
 	// usable by a loadbalancer, like http keep-alive.
 
+	// HTTP is a pointer pointing to a list of http selectors.
```
Comment (liggitt) [inline staging/src/k8s.io/api/extensions/v1beta1/types.go:735]: this is API-facing doc, so use the API field name (`http`) and don't mention pointer, which isn't meaningful in the REST API

Comment (wangyysde) [reply staging/src/k8s.io/api/extensions/v1beta1/types.go]: Thanks. I have modified them.

File: staging/src/k8s.io/api/extensions/v1beta1/types.go [hunk 3]
```diff
@@ -731,6 +734,13 @@ type IngressRuleValue struct {
 	// 2. Consider adding fields for ingress-type specific global options
 	// usable by a loadbalancer, like http keep-alive.
 
+	// http is point to a list of http selectors.
+	// The list is a collection of paths that map requests to backends.
```
Comment (liggitt) [inline staging/src/k8s.io/api/extensions/v1beta1/types.go:738]:
```suggestion
	// http is a list of http selectors pointing to backends.
```

**Comment by wangyysde** [comment]
Why is pull-kubernetes-verify alway  says api/openapi-spec out of date?  I have run hack/update-openapi-spec.sh several times on the local.  I have even deleted the files in the api/openapi-spec and run  hack/update-openapi-spec.sh again.  **You are apprieciated if you give me some advice!** Thanks.

**Comment by wangyysde** [comment]
Thanks! I found that the modification time of api/openapi-spec/swagger.json has be changed but the content of this file has not be changed  after I run hack/update-openapi-spec.sh on my local. I will try to check what are wrong on my local. Thanks again.

**Pull request merged**